analysis/analysis.py

```diff
@@ -43,6 +43,47 @@ def compute_total_view_time(student_data):
     return total_view_time
 
 
+def compile_response(student_data, question):
+    """
+    Returns answers as a dictionary with context as the keys and another dictionary containing each
+    response and their frequency as the value.
+    :param student_data: list of OrderedDicts, set of responses
+    :param question: string, question
+    :return: dictionary mapping strings to integers
+    """
+    data = {}
+    for elem in student_data:
+        student_question = elem['question']
+        student_response = elem['response'].lower()
+        question_context = elem['context']
+        if student_question == question:
+            if question_context not in data:
+                data[question_context] = {student_response: 1}
+            else:
+                if student_response in data[question_context]:
+                    data[question_context][student_response] += 1
+                else:
+                    data[question_context][student_response] = 1
+    return data
+
+
+def common_response(student_data, question, context):
+    """
+    Returns a list of the most common response(s) given a set of data, a question, and a context.
+    :param student_data: list of OrderedDicts, student response data
+    :param question: string, question
+    :param context: string, context
+    :return: list of strings
+    """
+    max_response = []
+    response_dict = compile_response(student_data, question)
+    responses_by_context = response_dict[context]
+    for response in responses_by_context:
+        if responses_by_context[response] == max(responses_by_context.values()):
```

Comment:
the value of ```max(responses_by_context.values())``` is common through all iterations through this loop, so factor it out rather than recalculating it each time